### PR TITLE
Add Functional tests

### PR DESCRIPTION
**Pull request by karelyatin**

### CODE REVIEW

**Review by gibizer — DISMISSED** [review]
I did not checked the actual test logic as that is neturon-operator specific I only looked at from the structure and best practices perspective

File: api/bases/neutron.openstack.org_neutronapis.yaml [hunk 1]
```diff
@@ -154,6 +154,26 @@ spec:
...
+                            of the Pod where this field is used. It makes that resource
+                            available inside a container.
+                          type: string
+                      required:
+                      - name
+                      type: object
+                    type: array
+                    x-kubernetes-list-map-keys:
+                    - name
+                    x-kubernetes-list-type: map
```
[the remark this answers is not in the document]
Comment (karelyatin) [reply api/bases/neutron.openstack.org_neutronapis.yaml:176]: Yes right, it's part of other PR https://github.com/openstack-k8s-operators/neutron-operator/pull/110, i just included here to get tests succeed, will be removed once other PR merges

File: test/functional/neutronapi_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+
+			scriptscm := th.GetConfigMap(
+				types.NamespacedName{
+					Namespace: neutronAPIName.Namespace,
+					Name:      fmt.Sprintf("%s-%s", neutronAPIName.Name, "scripts"),
+				},
+			)
+
+			Eventually(configData).ShouldNot(BeNil())
+			Eventually(scriptscm).ShouldNot(BeNil())
```
[the remark this answers is not in the document]
Comment (karelyatin) [reply test/functional/neutronapi_controller_test.go:481]: Yes right, updated

File: test/functional/neutronapi_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+			keystoneAPI := CreateKeystoneAPI(namespace)
+			DeferCleanup(DeleteKeystoneAPI, keystoneAPI)
+			configData := th.GetConfigMap(
+				types.NamespacedName{
+					Namespace: neutronAPIName.Namespace,
+					Name:      fmt.Sprintf("%s-%s", neutronAPIName.Name, "config-data"),
+				},
+			)
+
+			Eventually(configData).ShouldNot(BeNil())
```
[the remark this answers is not in the document]
Comment (karelyatin) [reply test/functional/neutronapi_controller_test.go:258]: Done

File: test/functional/neutronapi_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+			DeferCleanup(DeleteOVNDBClusters, dbs)
+			keystoneAPI := CreateKeystoneAPI(namespace)
+			DeferCleanup(DeleteKeystoneAPI, keystoneAPI)
+			configData := th.GetConfigMap(
+				types.NamespacedName{
+					Namespace: neutronAPIName.Namespace,
+					Name:      fmt.Sprintf("%s-%s", neutronAPIName.Name, "config-data"),
+				},
+			)
+			Eventually(configData).ShouldNot(BeNil())
```
[the remark this answers is not in the document]
Comment (karelyatin) [reply test/functional/neutronapi_controller_test.go:287]: Done

File: test/functional/base_test.go [hunk 5]
```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2022.
```
Comment (gibizer) [inline test/functional/base_test.go:2]: nit: 2023

**Review by gibizer — COMMENTED** [review]
Looks good to me. But I haven't checked it against the neutron-operator logic.

**Review by slawqo — COMMENTED** [review]
For me it looks ok but I would prefer that someone more experienced with tests in go will approve it

**Review by gibizer — APPROVED** [review]
Nice work @karelyatin ! I'm happy that there is yet another operator that has some functional test coverage.

**Pull request merged**